kamu-core/src/infra/utils/smart_transfer_protocol.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Arc;
+use url::Url;
+
+use crate::domain::{Dataset, SyncResult, SyncError, SyncListener};
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait SmartTransferProtocolClient: Sync + Send {
```

Comment:
We could probably even put this into `opendatafabric` crate as it's a part of the spec.